File: naiveBayes.r
install.packages('e1071',dependencies=TRUE)

library(e1071)

setwd("C:/Users/emili/Downloads/iris")
datos<- read.csv("iris.csv",sep = ";",dec=".",header = TRUE)
datos

#segmenta la tabla (dataset)

muestra<-sample(1:105,45)
muestra



ttesting<-datos[muestra,]
ttesting

taprendizje<-datos[-muestra,]
taprendizje
dim(taprendizje)

#sise capas ocultas
#rang pesos iniciales
#

modelo<-naiveBayes(tipo~.,data=taprendizje,size=6)
modelo


#Que tan bien predice el modelo
prediccion<-predict(modelo,ttesting[,-5])
prediccion

#Matriz de confusiOn
MC<-table(ttesting[,5],prediccion)
MC

#Porcentaje de buena clasificaciOn
acierto<-(sum(diag(MC)))/sum(MC)
acierto
error<-1-acierto
error

File: naiveBayes.py
# Set working directory and read data
import os
import pandas as pd


datos = pd.read_csv("iris.csv", sep=";", decimal=".", header=0)
print(datos)

# Segment the dataset
from sklearn.model_selection import train_test_split

X = datos.iloc[:, :-1]
y = datos.iloc[:, -1]

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=1)

# Create and train the model
from sklearn.naive_bayes import GaussianNB

modelo = GaussianNB()
modelo.fit(X_train, y_train)
print(modelo)

# Evaluate the model
y_pred = modelo.predict(X_test)
print(y_pred)

# Create confusion matrix
from sklearn.metrics import confusion_matrix

MC = confusion_matrix(y_test, y_pred)
print(MC)

# Calculate accuracy and error rate
acierto = sum([MC[i][i] for i in range(3)]) / sum(sum(MC))
print(acierto)
error = 1 - acierto
print(error)
